src/game/Entities/Unit.cpp

```diff
@@ -10414,16 +10419,17 @@ void Unit::SetHealthPercent(float percent)
     SetHealth(newHealth);
 }
 
-void Unit::SetPower(Powers power, uint32 val, bool withPowerUpdate /*= true*/)
+void Unit::SetPower(Powers power, float val, bool withPowerUpdate /*= true*/)
 {
-    if (GetPower(power) == val)
+    if (GetRealPower(power) == val)
```

Comment:
with float imprecision it might be best to keep that as
`if (GetPower(power) == uint32(val))`
so that tiny rounding errors don't cause the whole function to run every time

Not sure